requests/adapters.py

```diff
@@ -272,12 +272,10 @@ def request_url(self, request, proxies):
         :class:`HTTPAdapter <requests.adapters.HTTPAdapter>`.
 
         :param request: The :class:`PreparedRequest <PreparedRequest>` being sent.
-        :param proxies: A dictionary of schemes to proxy URLs.
+        :param proxies: A dictionary of schemes or schemes and hosts to proxy URLs.
         """
-        proxies = proxies or {}
-        scheme = urlparse(request.url).scheme
-        proxy = proxies.get(scheme)
-
+        proxy = select_proxy(request.url, proxies)
+        scheme = urlparse(request.url.lower()).scheme
         if proxy and scheme != 'https':
```

Comment:
How come we don't do `if proxy and scheme == 'http':`, like the documentation seems to indicate?